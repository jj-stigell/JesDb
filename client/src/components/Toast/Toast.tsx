import React from 'react'
import { ToastContainer, Slide } from 'react-toastify'
import { useThemeMode } from 'flowbite-react'

export default function Toast (): React.JSX.Element {
  const { computedMode } = useThemeMode()
  const width = window.innerWidth

  return (
    <ToastContainer
      position={width > 649 ? 'top-center' : 'bottom-center'}
      autoClose={3000}
      hideProgressBar={false}
      newestOnTop={false}
      closeOnClick
      rtl={false}
      pauseOnFocusLoss
      draggable
      pauseOnHover
      theme={computedMode}
      transition={Slide}
    />
  )
}
