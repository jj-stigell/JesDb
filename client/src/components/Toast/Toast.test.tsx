import '@testing-library/jest-dom'
import React from 'react'
import { render } from '@testing-library/react'
import { useThemeMode } from 'flowbite-react'

import Toast from './Toast'

// Mock useThemeMode and window.innerWidth
jest.mock('flowbite-react', () => ({
  useThemeMode: jest.fn()
}))

describe('Toast Component', () => {
  beforeEach(() => {
    (useThemeMode as jest.Mock).mockImplementation(() => ({ computedMode: 'light' }))
  })

  it('matches snapshot for large screen', () => {
    Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 650 })
    const { asFragment } = render(<Toast />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot for small screen', () => {
    Object.defineProperty(window, 'innerWidth', { writable: true, configurable: true, value: 649 })
    const { asFragment } = render(<Toast />)

    expect(asFragment()).toMatchSnapshot()
  })
})
