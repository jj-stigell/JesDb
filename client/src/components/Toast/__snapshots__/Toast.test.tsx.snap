// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Component matches snapshot for large screen 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`Toast Component matches snapshot for small screen 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
